modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -332,6 +353,31 @@ TensorMechanicsAction::actOutputGeneration()
   }
 }
 
+void
+TensorMechanicsAction::actGatherActionParameters()
+{
+  //
+  // Gather info about all other master actions when we add variables
+  //
+  if (_current_task == "validate_coordinate_systems" && getParam<bool>("add_variables"))
+  {
+    auto actions = _awh.getActions<TensorMechanicsAction>();
+    for (const auto & action : actions)
+    {
+      const auto size_before = _subdomain_id_union.size();
+      const auto added_size = action->_subdomain_ids.size();
+      _subdomain_id_union.insert(action->_subdomain_ids.begin(), action->_subdomain_ids.end());
```

Comment:
Interesting, any reason you don't use one of the STL "set" operations directly? That's what you are doing semantically.